src/main/java/org/mapstruct/intellij/codeinsight/references/MapstructNonNestedBaseReference.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at https://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.intellij.codeinsight.references;
+
+import com.intellij.codeInsight.lookup.LookupElement;
+import com.intellij.openapi.util.TextRange;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiType;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * A base reference to mapstruct annotations without nested types.
+ *
+ * @author Oliver Erhart
+ */
+public abstract class MapstructNonNestedBaseReference extends MapstructBaseReference {
```

Comment:
I tried to provide this kind of base reference on the level of `MapstructBaseReference` but it would have led to more duplication. Therefore I decided to provide a base reference on this level.

Reply:
I'm not sure that I understand what `MapStructNonNestedBaseReference` means.

Reply:
It feels to me like we need a type like `MapStructPropertyReference` which would be for the source and target references (which are the only ones where we have nested properties. I wouldn't mix the `nested` word though in the types, it's a bit strange to me